src/m365/planner/commands/task/task-add.ts

```diff
@@ -0,0 +1,394 @@
+    telemetryProps.bucketName = typeof args.options.bucketName !== 'undefined';
+    telemetryProps.startDateTime = typeof args.options.startDateTime !== 'undefined';
+    telemetryProps.dueDateTime = typeof args.options.dueDateTime !== 'undefined';
+    telemetryProps.percentComplete = typeof args.options.percentComplete !== 'undefined';
+    telemetryProps.assignedToUserIds = typeof args.options.assignedToUserIds !== 'undefined';
+    telemetryProps.assignedToUserNames = typeof args.options.assignedToUserNames !== 'undefined';
+    telemetryProps.description = typeof args.options.description !== 'undefined';
+    telemetryProps.orderHint = typeof args.options.orderHint !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this.getPlanId(args)
+      .then(planId => {
+        args.options.planId = planId;
+        return this.getBucketId(args, planId);
+      })
+      .then(bucketId => {
+        args.options.bucketId = bucketId;
+        return this.generateUserAssignments(args);
+      })
+      .then(assignments => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/tasks`,
+          headers: {
+            'accept': 'application/json;odata.metadata=none'
+          },
+          responseType: 'json',
+          data: {
+            planId: args.options.planId,
+            bucketId: args.options.bucketId,
+            title: args.options.title,
+            startDateTime: args.options.startDateTime,
+            dueDateTime: args.options.dueDateTime,
+            percentComplete: args.options.percentComplete,
+            assignments: assignments,
+            orderHint: args.options.orderHint
+          }
+        };
+
+        return request.post(requestOptions) as PlannerTask;
+      })
+      .then(newTask => { 
+        return this.updateTaskDetails(args.options, newTask);  
+      })
+      .then((res: any): void => {
+        logger.log(res);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getTaskDetailsEtag(taskId: string): Promise<string> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/tasks/${encodeURIComponent(taskId)}/details`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get(requestOptions)
+      .then((response: any) => {
+        const etag: string | undefined = response ? response['@odata.etag'] : undefined;
+
+        if (!etag) {
+          return Promise.reject(`Error fetching task details`);
+        }
+
+        return Promise.resolve(etag);
+      });
+  }
+
+  private updateTaskDetails(options: Options, newTask: PlannerTask): Promise<PlannerTask & PlannerTaskDetails> {
+    const taskId = newTask.id as string;
+
+    if (options.description) {
```

Comment:
To avoid indenting the whole method body, let's invert the clause to `if (!options.description)` with an early return. That way, we can make the code more readable.